src/app/storage/FabricTableImpl.hpp

```diff
@@ -0,0 +1,812 @@
+/**
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/storage/FabricTableImpl.h>
+#include <app/util/endpoint-config-api.h>
+#include <lib/support/DefaultStorageKeyAllocator.h>
+#include <lib/support/TypeTraits.h>
+#include <stdlib.h>
+
+namespace chip {
+namespace app {
+namespace Storage {
+
+using EntryIndex = Data::EntryIndex;
+
+/// @brief Tags Used to serialize entries so they can be stored in flash memory.
+/// kEntryCount: Number of entries in a Fabric
+/// kStorageIdArray: Array of StorageID struct
+/// kEndpointID: Tag for the Endpoint ID to that this entry applies to
```

Comment:
To answer your comment about casing...  There are places in this repo where we do ID, including in anything generated from the spec, but then all the code Google people wrote has Id because like you said that's the Google internal convention.

In practice it's a complete mess, but let's try to just have things consistent at least within a file.

That said, this comment does not seem to match the actual list of tags below anyway (e.g. there is no kEndpointID in the enum, but kEndpointEntryCount is missing from the docs here).

Reply:
Good catch, this was wrong before - updated to instead document kEndpointEntryCount